frontend/src/v5/ui/components/viewer/drawingViewer/drawingViewer.component.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { formatMessage } from '@/v5/services/intl';
+import { ToolbarButton } from '@/v5/ui/routes/viewer/toolbar/buttons/toolbarButton.component';
+import { ToolbarContainer, MainToolbar } from '@/v5/ui/routes/viewer/toolbar/toolbar.styles';
+import { useRef, useState } from 'react';
+import ZoomOutIcon from '@assets/icons/viewer/zoom_out.svg';
+import ZoomInIcon from '@assets/icons/viewer/zoom_in.svg';
+
+import { FileInputField } from '@controls/fileInputField/fileInputField.component';
+import { Button } from '@controls/button/button.component';
+import { FormattedMessage } from 'react-intl';
+import { SvgViewer } from './svgViewer.component';
+import { PanZoomHandler, centredPanZoom } from './panzoom/centredPanZoom';
+
+
+
+export const DrawingViewer = () => {
+	const [svgContent, setSvgContent] = useState('');
+	const [imgContent, setImgContent] = useState('');
+	const [zoomHandler, setZoomHandler] = useState<PanZoomHandler>();
+
+	const imgRef = useRef<HTMLImageElement | SVGSVGElement>();
+
+	const onClickButton = async ([file]: File[]) => {
+		if (file.type.includes('svg')) { 
+			setSvgContent(await file.text());
+			setImgContent('');
+		}
+
+		if (file.type.includes('png')) { 
+			var base64String = btoa(String.fromCharCode.apply(null, new Uint8Array(await file.arrayBuffer())));
+			setImgContent('data:image/png;base64,' + base64String);
+			setSvgContent('');
+		}
+	};
+
+	const onClickZoomIn = () => {
```

Comment:
What determines how far in you can zoom? On some images the zoom seems fine, but on others it will barely zoom.
Take for example this SVG of a Lystrosaurus. I am only able to zoom in once. The left side of the image shows the SVG at furthest zoom out whilst the right side shows the maximum extent of the zoom
![lystro](https://github.com/3drepo/3drepo.io/assets/62481250/c00e41fe-e94a-4f4b-85e6-1c0b274f7ebd)

Reply:
The original size of the image. For example: if its 10x10px . max zoom is 3, so max would be 30x30px but it also fits the  viewport with specific padding. 
@carmenfan  should I leave the original size of the image if its smaller than instead of the fixed padding? any idea on which criteria should I use on this case?

Reply:
As discussed on teams (just commenting here for completeness)

We can probably ignore this as it's very unlikely that in real life applications we'd run into a situation where the svg is tiny (we're talking about building or infrastructure 2D plans here) so we Santiago's assumption is probably good enough (until we're told otherwise by clients)